File: socketd-core/src/main/java/org/noear/socketd/protocol/Frame.java
package org.noear.socketd.protocol;

import org.noear.socketd.Message;

/**
 * 帧
 *
 * @author noear
 * @since 2.0
 */
public interface Frame {
    /**
     * 标志
     * */
    Flag getFlag();

    /**
     * 消息
     * */
    Message getMessage();
}

File: python/socket_websocket/socketd/core/module/Frame.py
from .Message import Message
from ..Costants import Flag


class Frame:

    def __init__(self, flag: Flag, message: Message):
        self.flag = flag
        self.message = message

    def get_flag(self) -> Flag:
        return self.flag

    def get_message(self) -> Message:
        return self.message

    def __str__(self) -> str:
        return f"Frame{{flag={self.flag}, message={self.message}}}"
